pippo-controller-parent/pippo-controller/src/main/java/ro/pippo/controller/extractor/FileItemExtractor.java

```diff
@@ -34,9 +37,25 @@ public boolean isApplicable(MethodParameter parameter) {
 
     @Override
     public Object extract(MethodParameter parameter, RouteContext routeContext) {
-        String name = parameter.getParameterName();
+        Param annotation = parameter.getAnnotation(Param.class);
```

Comment:
This might not work as expected - see `ControllerHandler.initExtractors()`. annotated FileItem might be processed by both extractors, and the result depends on extractors order.
If you want to reuse @Param annotation, the better way would be to remove FileItemExtractor completely and add additional check `if(FileItem.class.isAssignableFrom(parameterType))` in ParamExtractor.

Reply:
Sorry!
Thanks! Now I see the problem exactly where you said it. I believe that the solution you proposed is simple and makes sense.
I will do some tests in this direction...

Reply:
@whyicantusemyemailasusername , done. Any other suggestions? Please, let me know.